src/AppInstallerCLICore/Workflows/PortableFlow.cpp

```diff
@@ -363,7 +546,58 @@ namespace AppInstaller::CLI::Workflow
         // Reset termination to allow for ReportInstallResult to process return code.
         context.ResetTermination();
 
-        // TODO: create subcontext for uninstall
+        // Perform cleanup only if the install fails and is not an update.
+        const auto& installReturnCode = context.Get<Execution::Data::OperationReturnCode>();
+        bool isUpdate = WI_IsFlagSet(context.GetFlags(), Execution::ContextFlag::InstallerExecutionUseUpdate);
+
+        if (installReturnCode != 0 && installReturnCode != APPINSTALLER_CLI_ERROR_PORTABLE_PACKAGE_ALREADY_EXISTS && !isUpdate)
+        {
+            context.Reporter.Warn() << Resource::String::PortableInstallFailed << std::endl;
+            auto uninstallPortableContextPtr = context.CreateSubContext();
+            Execution::Context& uninstallPortableContext = *uninstallPortableContextPtr;
+            auto previousThreadGlobals = uninstallPortableContext.SetForCurrentThread();
+
+            uninstallPortableContext.Add<Execution::Data::PortableARPEntry>(context.Get<Execution::Data::PortableARPEntry>());
+            uninstallPortableContext << PortableUninstallImpl;
+        }
+    }
+
+    void PortableUninstallImpl(Execution::Context& context)
+    {
+        try
+        {
+            context.Reporter.Info() << Resource::String::UninstallFlowStartingPackageUninstall << std::endl;
+
+            context <<
+                RemovePortableExe <<
+                RemoveInstallDirectory <<
+                RemovePortableSymlink <<
+                RemovePortableARPEntry;
+
+            context.Add<Execution::Data::OperationReturnCode>(context.GetTerminationHR());
+        }
+        catch (...)
+        {
+            context.Add<Execution::Data::OperationReturnCode>(Workflow::HandleException(context, std::current_exception()));
+            AICLI_TERMINATE_CONTEXT(APPINSTALLER_CLI_ERROR_PORTABLE_UNINSTALL_FAILED);

```

Comment:
This seems like it shouldn't terminate the context, but continue on like install does.

Reply:
Removed termination